survey/dashboards/callbacks.py

```diff
@@ -0,0 +1,308 @@
+"""
+Dash callbacks for the survey dashboard
+
+Defines and registers the interactive callbacks of the dashboard's
+functionality, including filtering, metrics calculation, visualisations,
+and section summaries. All callbacks are registered to the provided dash_app instance.
+"""
+
+from dash.dependencies import Input, Output, State
+from dash import html, dcc
+from .utils import (
+    filter_survey_data,
+    calculate_section_averages,
+    create_section_figure,
+    calculate_section_statistics,
+    create_section_summary,
+    create_demographic_chart,
+    get_maturity_level
+)
+from .constants import MATURITY_LEVELS, COLOUR_PALETTE
+
+
+def register_callbacks(dash_app):
+    sections = dash_app.sections
+    demographic_fields = dash_app.demographic_fields
+
+    @dash_app.callback(
+        Output("stored-data", "data"),
+        [Input(f'{field["id"]}-filter', "value") for field in demographic_fields],
+    )
+    def update_stored_data(*filter_values):
+        all_responses = getattr(dash_app, 'survey_responses', [])
+
+        filters = {
+            field["id"]: value
+            for field, value in zip(demographic_fields, filter_values)
+            if value is not None
+        }
+
+        filtered_responses = filter_survey_data(all_responses, filters, dash_app)
+        section_averages = calculate_section_averages(filtered_responses, dash_app)
+
+        return {
+            "section_averages": section_averages,
+            "survey_responses": filtered_responses,
+        }
+
+    @dash_app.callback(
+        [Output(f'{field["id"]}-filter', "options") for field in demographic_fields],
+        [Input("stored-data", "data")],
+    )
+    def update_filter_options(data):
+        options = []
+
+        for field in demographic_fields:
+            if field["id"] == "age":
+                options.append([
+                    {"label": opt, "value": opt}
+                    for opt in ["Under 25", "25-34", "35-44", "45-54", "55-64", "65+"]
```

Comment:
Can we define these values in one place to avoid repetition?
![image](https://github.com/user-attachments/assets/c693653e-b2cd-432c-9848-ea08d2f088e0)